File: src/main/java/com/meetify/server/model/embeddable/Location.java
package com.meetify.server.model.embeddable;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import java.io.Serializable;

/**
 * com.meetify.server.model
 * Created by kr3v on 09.10.2016.
 */
@SuppressWarnings("unused")
@Embeddable
public class Location implements Serializable {
    @Column(name = "lat")
    private double lat;
    
    @Column(name = "lon")
    private double lon;
    
    public Location() {
        
    }
    
    public Location(double lat, double lon) {
        
        this.lat = lat;
        this.lon = lon;
    }
    
    public double getLon() {
        return lon;
    }
    
    public void setLon(double lon) {
        this.lon = lon;
    }
    
    public double getLat() {
        
        return lat;
    }
    
    public void setLat(double lat) {
        this.lat = lat;
    }
}

File: src/main/java/com/meetify/server/model/entity/Location.kt
package com.meetify.server.model.entity

import java.io.Serializable
import javax.persistence.Embeddable

/**
 * Цей клас є вбудовуваним, який відображає географічні координати певного об'єкту.
 * @author      Дмитро Байнак
 * @version     0.0.1
 * @since       0.0.1
 * @property    lat  широта.
 * @property    lon  довгота.
 * @constructor приймає 2 два числа з плаваючою комою.
 */
@Embeddable
data class Location(var lat: Double = 0.0, var lon: Double = 0.0) : Serializable